a3p-integration/proposals/a:upgrade-next/ante-fees.test.js

```diff
@@ -0,0 +1,72 @@
+import test from 'ava';
+
+import { CHAINID, GOV1ADDR, GOV2ADDR, agd } from '@agoric/synthetic-chain';
+
+test(`ante handler sends fee only to vbank/reserve`, async t => {
+  const [feeCollector, vbankReserve] = await Promise.all(
+    ['fee_collector', 'vbank/reserve'].map(async name => {
+      const {
+        account: {
+          '@type': moduleAcct,
+          base_account: { address },
+        },
+      } = await agd.query('auth', 'module-account', name);
+
+      t.is(
+        moduleAcct,
+        '/cosmos.auth.v1beta1.ModuleAccount',
+        `${name} is a module account`,
+      );
+      return address;
+    }),
+  );
+
+  const getBalances = () =>
+    Promise.all(
+      [feeCollector, vbankReserve].map(async address => {
+        const { balances } = await agd.query('bank', 'balances', address);
+        return balances;
+      }),
+    );
+
+  const [feeCollectorStartBalances, vbankReserveStartBalances] =
+    await getBalances();
+
+  const feeAmount = 999n;
+  const feeDenom = 'uist';
+  const result = await agd.tx(
+    `bank send ${GOV1ADDR} ${GOV2ADDR} 1234ubld --fees=${feeAmount}${feeDenom} \
+    --from=${GOV1ADDR} --chain-id=${CHAINID} --keyring-backend=test --yes`,
+  );
+  t.like(result, { code: 0 });
+
+  const [feeCollectorEndBalances, vbankReserveEndBalances] =
+    await getBalances();
+  t.deepEqual(
+    feeCollectorEndBalances,
+    feeCollectorStartBalances,
+    'fee_collector should not receive the fee',
+  );
+
+  let matched = false;
+  const vbankReserveExpectedBalances = vbankReserveStartBalances.map(
+    ({ amount, denom }) => {
+      if (denom === feeDenom) {
+        matched = true;
+        amount = String(BigInt(amount) + feeAmount);
+      }
+      return { amount, denom };
+    },
+  );
+  if (!matched) {
+    vbankReserveExpectedBalances.push({
+      amount: String(feeAmount),
+      denom: feeDenom,
+    });
+  }
```

Comment:
Alternative possibility:
```suggestion
  const feeDenomIndex =
    vbankReserveStartBalances.findIndex(({ denom }) => denom === feeDenom);
  const preFeeAmount = feeDenomIndex >= 0
    ? BigInt(vbankReserveStartBalances[feeDenomIndex].amount)
    : 0n;
  const beforeCount = feeDenomIndex >= 0 ? feeDenomIndex : Infinity;
  const vbankReserveExpectedBalances = [
    ...vbankReserveStartBalances.slice(0, beforeCount),
    { denom: feeDenom, amount: String(preFeeAmount + feeAmount) },
    ...vbankReserveStartBalances.slice(beforeCount),
  ];
```

Reply:
This needed a minor correction to use `beforeCount + 1` on line 60, and I opted for `vbankReserveStartBalances.length` because I can't wrap my head around `Infinity`. 😉 
But yes, this reads better.